src/bpftrace.cpp

```diff
@@ -1741,13 +1741,32 @@ uint64_t BPFtrace::resolve_cgroupid(const std::string &path) const
   return bpftrace_linux::resolve_cgroupid(path);
 }
 
+int sym_resolve_callback(const char *name, uint64_t addr, uint64_t, void *payload)
+{
+  struct bcc_symbol *sym = (struct bcc_symbol *)payload;
+  if (!strcmp(name, sym->name)) {
+    sym->offset = addr;
+    return -1;
+  }
+  return 0;
+}
+
 uint64_t BPFtrace::resolve_uname(const std::string &name, const std::string &path) const
 {
 #ifdef HAVE_BCC_ELF_FOREACH_SYM
-  bcc_symbol sym;
-  int err = bcc_resolve_symname(path.c_str(), name.c_str(), 0, 0, nullptr, &sym);
-  if (err)
+  struct bcc_symbol_option option = {
+    .use_debug_file = 0,
+    .check_debug_file_crc = 0,
+    .use_symbol_type = (1 << STT_OBJECT),
+  };
+
+  struct bcc_symbol sym;
+  sym.name = name.c_str();
+
+  int err = bcc_elf_foreach_sym(path.c_str(), sym_resolve_callback, &option, &sym);
```

Comment:
From reading the bcc code, I think bcc_elf_foreach_sym will still return 0 even if we never successfully resolve a symbol name. In this case, we'd be returning junk data from `sym.offset`, right?